storage/filesystem.go

```diff
@@ -165,6 +166,11 @@ func (f *Filesystem) RegisterMetadata(meta metadata.Metadata) (bool, error) {
 		return true, f.WriteMetadata(meta.VolumeID, meta)
 	}
 
+	if !apiequality.Semantic.DeepEqual(existingMeta.VolumeContext, meta.VolumeContext) {
+		f.log.WithValues("volume_id", meta.VolumeID).Info("volume context changed, updating file system metadata")
+		return true, f.WriteMetadata(meta.VolumeID, meta)
+	}
+
```

Comment:
I'm not sure I understand the purpose of this change - when/why could this metadata change? What case are we trying to handle here?

Reply:
Additionally: we should not overwrite the metadata file with `meta` (which likely won't contain any fields such as `nextIssuanceTime`) - instead, if we do want to support this, we should write `existingMetadata` back (after updating the context values). Otherwise we'll trigger an immediate re-issuance too.

Reply:
Thanks, I have added a commit which overwrites just the context to avoid re-issuance.

This change is needed since the Kubelet will [periodically call `NodePublishVolume` which renews any `TokenRequest` tokens](https://kubernetes-csi.github.io/docs/token-requests.html#usage). This is required so the driver always has a valid request token in the metadata when renewing a pods volume.